azurerm/internal/services/batch/resource_arm_batch_pool.go

```diff
@@ -380,6 +380,105 @@ func resourceArmBatchPool() *schema.Resource {
 					ValidateFunc: validate.NoEmptyStrings,
 				},
 			},
+			"network_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"subnet_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"endpoint_configuration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"inbound_nat_pools": {
```

Comment:
We could remove one level of schema by making this `enpoint_inbound_nat_pools` as there is nothing else for the endpoint?